django/db/backends/dummy/base.py

```diff
@@ -83,4 +83,4 @@ def __init__(self, *args, **kwargs):
         self.validation = BaseDatabaseValidation(self)
 
     def is_usable(self):
-        return True
+        return False
```

Comment:
There are at least 15 different places where ImproperyConfigured is raised in database backends other than for the dummy db. These are small, obscure misconfigurations. In each of these cases, checking the migrations would be silently skipped, without any clues for the user that a configuration that they may not even be aware of caused django to not check the migrations. It would be very difficult for the user to figure out what's going on in that case. In my opinion it would be preferable to go with changing the return value of is_usable on the dummy db and then documenting that migrations will not be verified if django cannot connect to the database. That way the user has a better understanding of what to expect.

I'm not sure what would be considered too invasive for this issue, but here's another idea:
https://github.com/lizlemon/django/commit/f39ec2d5a20719b1d560cfcd48cf706c5577e1c0
It's definitely more invasive than other ideas so far. It creates another db backend called 'empty' which is used just in case of a deliberately empty default database so that it does not raise ImproperlyConfigured, since it's a valid configuration. This might be useful for other things also, like this one:
https://code.djangoproject.com/ticket/26170